src/test/java/io/cdap/plugin/servicenow/source/ServiceNowRecordReaderTest.java

```diff
@@ -0,0 +1,123 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.servicenow.source;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.macro.Macros;
+import io.cdap.cdap.api.plugin.PluginProperties;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+public class ServiceNowRecordReaderTest {
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+  private ServiceNowSourceConfig serviceNowSourceConfig;
+  private ServiceNowRecordReader serviceNowRecordReader;
+
+  @Before
+  public void initializeTests() {
+    serviceNowSourceConfig = ServiceNowSourceConfigHelper.newConfigBuilder()
+      .setReferenceName("referenceName")
+      .setRestApiEndpoint("http://example.com")
+      .setUser("user")
+      .setPassword("password")
+      .setClientId("client_id")
+      .setClientSecret("client_secret")
+      .setTableNames("sys_user")
+      .setValueType("Actual")
+      .setStartDate("2021-12-30")
+      .setEndDate("2021-12-31")
+      .setTableNameField("tablename")
+      .build();
+
+    serviceNowRecordReader = new ServiceNowRecordReader(serviceNowSourceConfig);
+  }
+
+  @Test
+  public void testConstructor() throws IOException {
+    serviceNowRecordReader.close();
+    Assert.assertEquals(0, serviceNowRecordReader.pos);
+  }
+
+  @Test
+  public void testConstructor2() throws IOException {
+
+    serviceNowRecordReader.close();
+    Assert.assertEquals(0, serviceNowRecordReader.pos);
+    Assert.assertEquals("tablename", serviceNowSourceConfig.getTableNameField());
+    Assert.assertEquals("tablename", serviceNowSourceConfig.getTableName());
+    Assert.assertEquals("2021-12-30", serviceNowSourceConfig.getStartDate());
+    Assert.assertEquals("referenceName", serviceNowSourceConfig.getReferenceName());
+    Assert.assertEquals("2021-12-31", serviceNowSourceConfig.getEndDate());
+    PluginProperties properties = serviceNowSourceConfig.getProperties();
+    Assert.assertEquals("PluginProperties{properties={}, macros=Macros{lookupProperties=[], macroFunctions=[]}}",
+                        properties.toString());
+    Assert.assertTrue(properties.getProperties().isEmpty());
+    Macros macros = properties.getMacros();
+    Assert.assertEquals("Macros{lookupProperties=[], macroFunctions=[]}", macros.toString());
+    Assert.assertTrue(macros.getMacroFunctions().isEmpty());
+    Assert.assertTrue(macros.getLookups().isEmpty());
+  }
+
+  @Test
+  public void testConvertToValue() {
+    Schema fieldSchema = Schema.of(Schema.LogicalType.TIMESTAMP_MILLIS);
+    thrown.expect(IllegalStateException.class);
+    serviceNowRecordReader.convertToValue("Field Name", fieldSchema, new HashMap<>(1));
+  }
+
+  @Test
+  public void testConvertToValue2() {
+    // TODO: This test is incomplete.
```

Comment:
address TODO?

Reply:
This has been removed. Will be taken care of, in the next PR for service-now JUnits.